### DISCUSSION TITLE
Arduino help wanted with Delta Shop Air Cleaner remote

### DISCUSSION THREAD

**Discussion by papavienneau** [Arduino-IRremote/Arduino-IRremote#1124] [Q&A] [answered]
Help wanted with Delta Shop Air Cleaner. Broke the IR remote for my cleaner and all stores listing parts for these units show the remote as being “obsolete, no longer available”. I would like to build a working remote, first testing it using an UNO, to transmit all 3 functions (ON/OFF, Speed and Timer) to my Air Cleaner. Stores listing parts for the all Delta 3 speed unit, i.e. 50-875, 50-868, etc… show the same part number 907382 as the remote for all Delta 3 speed units.
A gentleman from Huston Texas with same problem approached the JP1 Remotes group @ [www.hifi-remote.com](http://www.hifi-remote.com/) and with the help of membership, was able to program three keys of a multi button remote (URC-881x ) to control his Shop Air Cleaner. A file relating to the URC-881x can be downloaded from this site an gives the three Hex codes for IR remote as 81 for ON/OFF, 82 for SPEED and 88 for TIMER. 
In desperation, my Air Cleaner is mounted from the shop ceiling and a ladder was required to reach its manual control, I tried every IR remote I could find, every button on every remote, and succeeded in turning the air cleaner on and off using a house fan remote.  I then used the IRremote library “receive dump” to generate the following: 

 
115200 baud
 

Protocol=PulseWidth Raw-Data=0x40D 11 bits LSB first

Send with: IrSender.sendPulseDistanceWidth(38, 1250, 400, 1250, 450, 400, 1250, 0x40D, 11, PROTOCOL_IS_LSB_FIRST, <RepeatPeriodMillis>, <numberOfRepeats>);

Raw result in internal ticks (50 us) - with leading gap
rawData[24]: 
 -65535
 +25,- 8
 +25,- 9 + 8,-25 +25,- 8 +25,- 9
 + 8,-25 + 8,-26 + 8,-25 + 8,-25
 + 9,-25 + 8,-25 +25
Sum: 392
Raw result in microseconds - with leading gap
rawData[24]: 
 -3276750
 +1250,- 400
 +1250,- 450 + 400,-1250 +1250,- 400 +1250,- 450
 + 400,-1250 + 400,-1300 + 400,-1250 + 400,-1250
 + 450,-1250 + 400,-1250 +1250
Sum: 19600

Result as internal 8bit ticks (50 us) array - compensated with MARK_EXCESS_MICROS=20
uint8_t rawTicks[23] = {25,8, 25,9, 8,25, 25,8, 25,9, 8,25, 8,26, 8,25, 8,25, 9,25, 8,25, 25};  // Protocol=PulseWidth Raw-Data=0x40D 11 bits LSB first

Result as microseconds array - compensated with MARK_EXCESS_MICROS=20
uint16_t rawData[23] = {1230,420, 1230,470, 380,1270, 1230,420, 1230,470, 380,1270, 380,1320, 380,1270, 380,1270, 430,1270, 380,1270, 1230};  // Protocol=PulseWidth Raw-Data=0x40D 11 bits LSB first

uint16_t address = 0x0;
uint16_t command = 0x0;
uint32_t rawData = 0xON;

Pronto Hex as string
char prontoData[] = "0000 006D 000C 0000 0031 000F 0031 0011 0010 002F 0031 000F 0031 0011 0010 002F 0010 0031 0010 002F 0010 002F 0012 002F 0010 002F 0031 06C3 ";
006D0C00310F3111102F310F3111102F1031102F102F122F102F31 06C3

Protocol=PulseWidth Repeat gap=7250us Raw-Data=0x40D 11 bits LSB first

Interestingly, the Air cleaner is looking for an 8 bit word to turn ON/OFF HEX81= 10000001 and it is responding to an 11 bit word 40D which is equal to 10000001101 that has exactly the same leading 8 bits.
This where I am stuck.
Perhaps a member of this community is also familiar with JP1 Remotes and can help create a sketch allowing an UNO to control this Air Cleaner.  Alternatively, perhaps someone has access to an original Delta remote or a URC remote programed as noted by JP1 and could use the IRremote library “receive dump” to generate the 3 required commands.
Papavienneau

**Comment by bengtmartensson** [comment, marked as answer]
Hi. I am known as Barf in the JP1 remotes forum. The said files appers to be [this](http://www.hifi-remote.com/forums/dload.php?action=file&file_id=26460). So download that, and load it into the RemoteMaster program (as described in the JP1 forum), Then, with RemoteMaster (reasonably recent version), export is as a Girr file. This is an XML file, usable both for human reading and automatic pre-processing. You will find that all of the three commands has a repeat- and an ending sequence. The repeat sequence is sent as long as you "hold down the button" (literally or virtually), while the ending sequence is sent once when you "release the button". The sequence consists of on- and off-periods in micro seconds (with + and + signs respectively, remove for the Arduino sketch).

How to send raw signals is shown in the SendRawDemo.ino. You should now be able to translate the sequences to an Arduino sketch for the current library. 

If you need more help than that, just say so.

**Reply by papavienneau** [reply]
Bengt
Got it working!  I am not familiar with programing both the repeat and ending sequences on an Arduino so I only used the repeat sequences for all three functions.  I individually programmed each function and simply set the Arduino to repeat sending the resulting code continuously every 300 milliseconds.  Using a cheap (about $4.00 US) 6 button learning IR remote, I programed its ON/OFF button as HEX 81, both the Volume UP and DOWN buttons as HEX 82 for Speed and both Channel UP and DOWN buttons as HEX 88 for the Timer function.  Then, simply left the 6th button (the Mute function) unprogrammed.  It works great!  Thanks a lot.
Eric

**Comment by papavienneau** [comment]
Bengt

Thanks for the info regarding the Delta 50-875 remote. I am new to Arduino
and will certainly give this a shot.  May take you up on more help.  As a
76 year old, I am trying to avoid climbing a ladder to control my air
cleaner and as a retired Electrical Engineer, I enjoy playing with
electronics.  Will let you know how I make out.

Eric (papavienneau)
